tomviz/DataSource.cxx

```diff
@@ -123,6 +127,26 @@ void deserializeDataArray(const pugi::xml_node& ns, vtkDataArray* array)
 }
 }
 
+DataSource::ImageFuture::ImageFuture(Operator* op,
+                                     vtkSmartPointer<vtkImageData> imageData,
+                                     PipelineWorker::Future* future,
+                                     QObject* parent)
+  : QObject(parent), m_operator(op), m_imageData(imageData), m_future(future)
+{
+
+  if (m_future != nullptr) {
+    connect(m_future, SIGNAL(finished()), this, SIGNAL(finished()));
+    connect(m_future, SIGNAL(canceled()), this, SIGNAL(canceled()));
+  }
+}
+
+DataSource::ImageFuture::~ImageFuture()
+{
+  if (this->m_future != nullptr) {
```

Comment:
Yeah, clang-style wouldn't do this for me, I would rather move to m_ and that should replace this->, let's not use both, I could probably do a quick grep for this->m_ to m_, at some point soon.